### ISSUE TITLE
Function pointer calls where pointer is dereferenced are rewritten incorrectly

### ISSUE THREAD

**Issue by randomPoison** [immunant/IA2-Phase2#410]
If you have a function pointer `fn_ptr`, you can optionally dereference that pointer when making the call, i.e. you can do both `fn_ptr()` and `(*fn_ptr)()`. The former case is currently rewritten correctly, but the latter case fails to compile when rewritten. This is because once the function pointer is replaced the dereference operation fails, i.e. we get `IA2_CALL((*fn_struct))()`.

It would be better if the rewriter could detect this extraneous dereference and remove it in the rewritten code.

**Comment by ayrtonm** [comment]
yeah this is a known issue. I thought there was an issue or a comment in a test mentioning this, but iirc the extra `*` isn't easy to detect without forcing the rewriter to expect a very specific regex for the callee expression.

One idea I haven't tried though is checking how it shows up in `clang -ast-dump` and changing the AST matchers. I don't think it'll be easy though.
